cmd/cmd.go

```diff
@@ -348,6 +360,13 @@ func (o *options) generateTemplate() (*template.Template, error) {
 			}
 			return string(b), nil
 		},
+		"tryParseJSON": func(text string) map[string]interface{} {
+			obj := make(map[string]interface{})
+			if err := json.Unmarshal([]byte(text), &obj); err != nil {
+				return nil
+			}
+			return obj
+		},
```

Comment:
```suggestion
		"tryParseJSON": func(text string) map[string]interface{} {
			decoder := json.NewDecoder(strings.NewReader(text))
			decoder.UseNumber()
			obj := make(map[string]interface{})
			if err := decoder.Decode(&obj); err != nil {
				return nil
			}
			return obj
		},
```

Though parseJSON has the same issue, it seems that tryParseJSON does not parse large numbers such as timestamps as expected.

Let's say we have logs below. The timestamp `1257894000` is `2009-11-10 23:00:00` in UTC.

```
{"level":"info","ts":1257894000,"caller":"sandbox2210433069/prog.go:10","msg":"hello","key1":"val1"}
{"level":"info","ts":1257894000,"caller":"sandbox2210433069/prog.go:13","msg":"world","key1":"val1"}
```

When we use `tryParseJSON` and `formatTsRFC3339Nano`, we will get `0001-01-01T00:00:00Z`, which is incorrect.

```
zap-log c [0001-01-01T00:00:00Z] info (sandbox2210433069/prog.go:10) hello
zap-log c [0001-01-01T00:00:00Z] info (sandbox2210433069/prog.go:13) world
```

Please refer to https://stackoverflow.com/questions/22343083/json-unmarshaling-with-long-numbers-gives-floating-point-number for details.

Reply:
Thank you for the suggestion, I didn't realise that `UseNumber` is not a default option!